src/Library/demos/Memory Monitor/main.vala

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+private HashTable<string, int?> cache;
+
+public void main () {
+  cache = new HashTable<string, int?> (string.hash, str_equal);
+  cache["a"] = 1;
+  cache["b"] = 2;
+  cache["c"] = 3;
+
+  var memory_monitor = MemoryMonitor.dup_default ();
+  memory_monitor.low_memory_warning.connect ((level) => {
+    // Use inequalities for checking as new levels may be added in the future
```

Comment:
I am not sure if inequalities are appropriate here. Its not immediately understandable, what the inequality means here, and people shouldnt need to look up the API docs for the enum, to see what other entries exist. Also if its just `==`'s then, maybe a switch statement would be better suited? Not sure...
The switch statement could be missing break's to still print the other warnings.

Reply:
It is a recommendation from the enum docs: https://docs.gtk.org/gio/enum.MemoryMonitorWarningLevel.html. I agree that it isn't obvious, so a link to Valadoc would be useful in this case.